contributor-guide/modules/ROOT/pages/design-guide/design-guide.adoc

```diff
@@ -100,6 +100,20 @@ Make sure your code compiles in the presence of the `min()` and `max()` macros.
 
 * If you want to declare or define a function or a member function named min or max, then you must use the `BOOST_PREVENT_MACRO_SUBSTITUTION` macro. Instead of writing `int min() { return 0; }` you should write int min `BOOST_PREVENT_MACRO_SUBSTITUTION () { return 0; }` This is true regardless if the function is a free (namespace scope) function, a member function or a static member function, and it applies for the function declaration as well as for the function definition.
 
+== Support for Dynamic Libraries
+
+A dynamic library is a compiled collection of binary code and data that can be loaded and linked into a program at runtime, allowing shared use of library functions across multiple applications without the need for recompilation. Unlike static libraries, which are linked at compile time and become part of the resulting executable, dynamic libraries remain separate files and are often employed to save memory, reduce redundancy, and facilitate modular application design.
+
+This concept is known by different names on various platforms: shared objects (.so) on UNIX systems, dynamic link libraries (.dll) on Windows, and dynamic shared libraries (.dylib) on macOS.
+
+If your plan for your library is to enable its use as a dynamic library, you need to implement two sets of macros:
```

Comment:
I'm not sure if providing the functionality is optional. I'm not saying it's not. I really don't know.

> you need to implement two sets of macros:

The macros are implemented. They just need to use them.